Added xresources patch from dmenu to support defining font sizes

ppenguin opened the pull request:
Added xresources patch from `dmenu` to support defining font sizes (and… colours) for instantmenu in ~/.Xresources.

To have any effect for e.g. `instanthotkeys`, this should be combined with the PR which removes the hardcoded font size from `instanthotkeys`.

The modification is based on the patch https://tools.suckless.org/dmenu/patches/xresources/, but was only tested for the `dmenu.font` property.

(Since my `C` is quite rusty, maybe the code should be reviewed, especially because the pointer handling of the patch was a bit patchy to begin with).

ppenguin committed:
Added xresources patch from dmenu to support defining font sizes (and colours) for instantmenu in ~/.Xresources

con-f-use commented:
@paperbenni Can you have a look? I'm very much in favor of this one.

con-f-use commented on instantmenu.c:
```diff
@@ -1632,6 +1644,39 @@ usage(void)
+			colors[SchemeSel][ColBg] = strdup(xval.addr);
+		else
+			colors[SchemeSel][ColBg] = strdup(colors[SchemeSel][ColBg]);
+		if (XrmGetResource(xdb, "dmenu.selforeground", "*", &type, &xval))
```
Maybe the `dmenu.<property>` keys here like `demnu.selforeground` shloud all be `instantmenu.<property>`. After all instantMENU is it's own thing and some people might install dmenu and instantMENU side by side, wishing to customize them separately. Also instantWM has it's own `instantwm.font`, that isn't `dwm.font` either.

paperbenni replied:
I agree, instantmenu should be configured separately from dmenu. This makes it more consistent since instantmenu might also get options that dmenu has not and allows it to deviate from dmenu without breaking one of the two if both are installed

paperbenni commented:
I think this patch needs some work and can't be carried over as is from dmenu. instantMENU has a lot more colors than dmenu, some of which go black with this patch. It also has them in the wrong order, the background color for instance is a lot lighter with the build from this pr. Fonts work a little different from dmenu as well, I'll have to test if this interferes with that in any way.

uvera commented on instantmenu.c:
```diff
@@ -1422,8 +1427,15 @@ setup(void)
-	for (j = 0; j < SchemeLast; j++)
-		scheme[j] = drw_scm_create(drw, colors[j], 3);
+ 	for (j = 0; j < SchemeLast; j++) {
+		scheme[j] = drw_scm_create(drw, (const char**)colors[j], 2);
```
last param of the drw_scm_create should be 3 to factor in ColDetail

con-f-use replied:
```suggestion
		scheme[j] = drw_scm_create(drw, (const char**)colors[j], 3);
```
Like @uvera suggested.

ppenguin committed:
Update instantmenu.c

Modify accotding to https://github.com/instantOS/instantMENU/pull/8/files/a13a15260f4daba08708cdf33788a707cae2d017#r568604665

Co-authored-by: con-f-use <con-f-use@users.noreply.github.com>

ppenguin commented:
Oops, I missed #9, since it appears @uvera actually knows what he's doing (:wink: ), probably best close #8 and merge #9?
Any open points on #9?

paperbenni commented:
I merged #9
